kythe/cxx/indexer/proto/file_descriptor_walker.cc

```diff
@@ -297,6 +301,81 @@ void FileDescriptorWalker::VisitImports() {
   }
 }
 
+namespace {
+std::string SignAnnotation(
+    google::protobuf::GeneratedCodeInfo::Annotation annotation) {
+  std::string signature;
+  std::stringstream sig(signature);
+  bool first_node = true;
+  for (const auto& node : annotation.path()) {
+    sig << (first_node ? "" : ".") << node;
+    first_node = false;
+  }
+  return sig.str();
+}
+
+VName VNameForAnnotation(
+    VName context_vname,
+    google::protobuf::GeneratedCodeInfo::Annotation annotation) {
```

Comment:
Both of these arguments should be const-ref instead of values.

Reply:
done